### PR TITLE
BUG: Fix field's appearance problem

### PR DESCRIPTION
**Pull request by liuzhuoling2011**
I have tried many python pdf-related tools, none of them work correctly for me.
Finally I found this, it meet all the needs but have **some small problem** such as: 
1. data filled in can't be seen until you click the field (appearance problem)
2. check box can't be selected

to make better user experience, I made some change~
now, use this library is more easier, just need a little customer code!

```python
from PyPDF2 import PdfFileWriter, PdfFileReader

def fill_pdf_with_dict(input_file, output_file, dict_data):
    pdf_reader = PdfFileReader(input_file, strict=False)
    pdf_writer = PdfFileWriter()

    for i in range(pdf_reader.getNumPages()):
        page = pdf_reader.getPage(i)
        pdf_writer.updatePageFormFieldValues(page, dict_data[i])
        pdf_writer.addPage(page)

    with open(output_file, 'wb') as outputStream:
        pdf_writer.write(outputStream)
```

### CODE REVIEW

**Comment by mwhit74** [comment]
A fix for the appearance was started a while ago (#412) , just never got incorporated into the code base. I don't think it is significantly than the fix given here. 

I am confused why the /NeedAppearances is updated in the PdfFileReader class in this solution.

File: PyPDF2/pdf.py [hunk 1]
```diff
@@ -379,13 +390,15 @@ def updatePageFormFieldValues(self, page, fields):
...
-                    })
+                    if writer_annot.get('/FT') == '/Btn':
+                        writer_annot.update({
+                            NameObject("/V"): NameObject(fields[field]),
+                            NameObject("/AS"): NameObject(fields[field])
+                        })
+                    else:
+                        writer_annot.update({
+                            NameObject("/V"): TextStringObject(fields[field])
+                        })
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:401]:
```suggestion
                        })
                elif writer_parent_annot.get('/T') == field:
                    writer_parent_annot.update({
                        NameObject("/V"): TextStringObject(fields[field])
                    })
```

File: PyPDF2/pdf.py [hunk 2]
```diff
@@ -379,14 +390,20 @@ def updatePageFormFieldValues(self, page, fields):
...
                 elif writer_parent_annot.get('/T') == field:
                     writer_parent_annot.update({
                         NameObject("/V"): TextStringObject(fields[field])
                     })
-
+
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:406]:
```suggestion

```

**Comment by MartinThoma** [comment]
It seems as if this PR breaks something

File: PyPDF2/pdf.py [hunk 3]
```diff
@@ -119,6 +119,17 @@ def __init__(self):
...
+            self._root_object["/AcroForm"].update({
+                NameObject("/NeedAppearances"): BooleanObject(True)
+            })
+        else:
+            self._root_object.update({
+                NameObject("/AcroForm"): IndirectObject(len(self._objects), 0, self)
+            })
+            need_appearances = NameObject("/NeedAppearances")
+            self._root_object["/AcroForm"][need_appearances] = BooleanObject(True)
+
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:132]:
```suggestion
```

Doing this caused an issue; see https://github.com/py-pdf/PyPDF2/pull/970

File: PyPDF2/pdf.py [hunk 4]
```diff
@@ -1195,6 +1200,11 @@ def _showwarning(message, category, filename, lineno, file=warndest, line=None):
 
         self._override_encryption = False
 
+        if "/AcroForm" in self.trailer["/Root"]:
+            self.trailer["/Root"]["/AcroForm"].update({
+                NameObject("/NeedAppearances"): BooleanObject(True)
+            })
+
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:1207]:
```suggestion
```

We now have `writer.set_need_appearances_writer()` which is used in `writer.update_page_form_field_values`

**Comment by MartinThoma** [comment]
Note for myself:

> AS: The annotation’s appearance state, which selects the applicable
appearance stream from an appearance subdictionary

**Pull request closed without merging**